pkg/istio-agent/xds-proxy.go

```diff
@@ -166,12 +168,20 @@ func (p *XdsProxy) StreamAggregatedResources(downstream discovery.AggregatedDisc
 		select {
 		case err := <-upstreamError:
 			// error from upstream Istiod.
-			proxyLog.Infof("Upstream terminated with status %v", err)
+			if isExpectedGRPCError(err) {
+				proxyLog.Debugf("upstream terminated with status %v", err)
+			} else {
+				proxyLog.Warnf("upstream terminated with unexpected error %v", err)
+			}
 			_ = upstream.CloseSend()
 			return err
 		case err := <-downstreamError:
 			// error from downstream Envoy.
-			proxyLog.Infof("downstream terminated with status %v", err)
+			if isExpectedGRPCError(err) {
```

Comment:
Now, i think downstream(envoy) will not reconnect periodically

Reply:
why? I just changed the log level

Reply:
isn't the connection long running?

Reply:
Yes it is. But I did not change those semantics now. I just condition and changed the log level. May be I am not understanding what you are saying - has this PR changed any behaviour wrt to that?